src/material-experimental/mdc-select/harness/select-harness.ts

```diff
@@ -0,0 +1,122 @@
+ * @dynamic
+ */
+export class MatSelectHarness extends ComponentHarness {
+  private _documentRootLocator = this.documentRootLocatorFactory();
+  private _panel = this._documentRootLocator.locatorFor(PANEL_SELECTOR);
+  private _backdrop = this._documentRootLocator.locatorFor('.cdk-overlay-backdrop');
+  private _optionalPanel = this._documentRootLocator.locatorForOptional(PANEL_SELECTOR);
+  private _options = this._documentRootLocator.locatorForAll(`${PANEL_SELECTOR} .mat-option`);
+  private _groups = this._documentRootLocator.locatorForAll(`${PANEL_SELECTOR} .mat-optgroup`);
+  private _trigger = this.locatorFor('.mat-select-trigger');
+  private _value = this.locatorFor('.mat-select-value');
+
+  static hostSelector = '.mat-select';
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a select with
+   * specific attributes.
+   * @param options Options for narrowing the search.
+   * @return `HarnessPredicate` configured with the given options.
+   */
+  static with(options: SelectHarnessFilters = {}): HarnessPredicate<MatSelectHarness> {
+    return new HarnessPredicate(MatSelectHarness)
+        .addOption('id', options.id, async (harness, id) => {
+          const harnessId = (await harness.host()).getAttribute('id');
+          return (await harnessId) === id;
+        });
+  }
+
+  /** Gets a boolean promise indicating if the select is disabled. */
+  async isDisabled(): Promise<boolean> {
+    return (await this.host()).hasClass('mat-select-disabled');
+  }
+
+  /** Gets a boolean promise indicating if the select is valid. */
+  async isValid(): Promise<boolean> {
+    return !(await (await this.host()).hasClass('ng-invalid'));
+  }
+
+  /** Gets a boolean promise indicating if the select is required. */
+  async isRequired(): Promise<boolean> {
+    return (await this.host()).hasClass('mat-select-required');
+  }
+
+  /** Gets a boolean promise indicating if the select is empty (no value is selected). */
+  async isEmpty(): Promise<boolean> {
+    return (await this.host()).hasClass('mat-select-empty');
+  }
+
+  /** Gets a boolean promise indicating if the select is in multi-selection mode. */
+  async isMultiple(): Promise<boolean> {
+    const ariaMultiselectable = (await this.host()).getAttribute('aria-multiselectable');
+    return (await ariaMultiselectable) === 'true';
+  }
+
+  /** Gets a promise for the select's value text. */
+  async getValueText(): Promise<string> {
+    return (await this._value()).text();
+  }
+
+  /** Focuses the select and returns a void promise that indicates when the action is complete. */
+  async focus(): Promise<void> {
+    return (await this.host()).focus();
+  }
+
+  /** Blurs the select and returns a void promise that indicates when the action is complete. */
+  async blur(): Promise<void> {
+    return (await this.host()).blur();
+  }
+
+  /** Gets the select panel. */
+  async getPanel(): Promise<TestElement> {
```

Comment:
is there a reason to expose this element? what would a user do with it in a test?

Reply:
You can use it to look at the scroll position and the position of the dropdown itself. We have a bunch of our own tests that deal with the dropdown.